src/main/java/com/salesforce/storm/spout/dynamic/consumer/Record.java

```diff
@@ -43,13 +44,18 @@
      * @param offset offset of the record.
      * @param values values of the record.
      */
-    public Record(final String namespace, final int partition, final long offset, final Values values) {
+    public Record(final String consumerId, final String namespace, final int partition, final long offset, final Values values) {
```

Comment:
Is this consumerId or VirtualSpout Identifier?

Reply:
This is the consumer id. We don't pass the vspout identifier into the consumer, we just `toString()` it. I believe we erred on that path so as to allow ourselves the ability to fully split the consumer out. I think the bigger question, from my stand point, is do we expect the Consumer interface itself to ever live outside of the DynamicSpout? I'm not sure that we do...  Maybe we split Kafka off as a separate module, but does that mean the interface can be used outside of the DynamicSpout?